test/app-tap/CMakeLists.txt

```diff
@@ -6,3 +6,103 @@ target_link_libraries(module_api msgpuck)
 set(CMAKE_C_FLAGS "-Wall -Wextra -std=c99")
 set(CMAKE_C_FLAGS_DEBUG "-g -O0 -Werror")
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2")
+
+make_lua_path(LUA_PATH
+  PATHS
+    ./?.lua
+    ${CMAKE_CURRENT_SOURCE_DIR}/lua/?.lua
+    # Needed for test-run/test_run.lua.
+    ${PROJECT_SOURCE_DIR}/test-run/?.lua
+)
+
+make_lua_path(LUA_CPATH
+  PATHS
+  ${CMAKE_CURRENT_BINARY_DIR}/?.so
+)
+
+set(TEST_SUITE_NAME "app-tap")
+set(TARANTOOL_BIN $<TARGET_FILE:tarantool>)
+
+message(STATUS "Add test suite ${TEST_SUITE_NAME}")
+
+# XXX: The call produces both test and target <app-tap-deps>
+# as a side effect.
+_add_test_suite_target(${TEST_SUITE_NAME}
+  LABELS ${TEST_SUITE_NAME}
+  DEPENDS module_api
+)
+
+# XXX: Setting PATH is required by app-tap/minimal.test.lua.
+# I would pass TARANTOOL_BIN instead and update the test.
+set(ENV_PATH $ENV{PATH})
+set(ENV_PATH "${PROJECT_BINARY_DIR}/src:${ENV_PATH}")
+list(APPEND TEST_ENV "LUA_PATH=${LUA_PATH}")
+list(APPEND TEST_ENV "LUA_CPATH=${LUA_CPATH}")
+list(APPEND TEST_ENV "PATH=${ENV_PATH}")
+
+file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR} *.test.lua)
+foreach(test_path ${tests})
+  get_filename_component(test_name ${test_path} NAME)
+  # FIXME: By default, GLOB lists directories.
+  # Directories are omitted in the result if LIST_DIRECTORIES
+  # is set to false. New in version CMake 3.3.
+  if(${test_name} STREQUAL ${TEST_SUITE_NAME})
+    continue()
+  endif()
+  set(test_title "test/${TEST_SUITE_NAME}/${test_name}")
+  set(work_dir "${CMAKE_CURRENT_BINARY_DIR}/${test_name}-workdir")
+  file(MAKE_DIRECTORY ${work_dir})
+  set(TEST_CMD "${RM} -f ${work_dir}/* && ${TARANTOOL_BIN} ${test_path}")
+  add_test(NAME ${test_title}
+           COMMAND ${BASH} -c "${TEST_CMD}"
+           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+  list(APPEND TEST_ENV "TT_WORK_DIR=${work_dir}")
+  set_tests_properties(${test_title} PROPERTIES
+    ENVIRONMENT "${TEST_ENV}"
+    LABELS "${TEST_SUITE_NAME};integration"
+    DEPENDS ${TEST_SUITE_NAME}-deps
+    FAIL_REGULAR_EXPRESSION "${CTEST_FAIL_REGEXP}"
+  )
+endforeach()
+
+if (CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
```

Comment:
Minor: `if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")` looks simplier to me.
Here and below.

Reply:
Fixed:

```diff
diff --git a/test/app-tap/CMakeLists.txt b/test/app-tap/CMakeLists.txt
index d129178bf9..15c3274a82 100644
--- a/test/app-tap/CMakeLists.txt
+++ b/test/app-tap/CMakeLists.txt
@@ -68,7 +68,7 @@ foreach(test_path ${tests})
   )
 endforeach()
 
-if (CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
+if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
   set_tests_properties(
     # See test/app-tap/suite.ini.
     test/app-tap/gh-2717-no-quit-sigint.test.lua
diff --git a/test/box-luatest/CMakeLists.txt b/test/box-luatest/CMakeLists.txt
index c8e4dd7297..930bb7c377 100644
--- a/test/box-luatest/CMakeLists.txt
+++ b/test/box-luatest/CMakeLists.txt
@@ -57,7 +57,7 @@ foreach(test_path ${tests})
   )
 endforeach()
 
-if (CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
+if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
   set_tests_properties(
     # See test/box-luatest/suite.ini.
     test/box-luatest/gh_6819_iproto_watch_not_implemented_test.lua
diff --git a/test/box-tap/CMakeLists.txt b/test/box-tap/CMakeLists.txt
index 46ab35d84a..81182886f2 100644
--- a/test/box-tap/CMakeLists.txt
+++ b/test/box-tap/CMakeLists.txt
@@ -42,7 +42,7 @@ foreach(test_path ${tests})
   )
 endforeach()
 
-if (CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
+if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
   set_tests_properties(
     # See test/box-tap/suite.ini.
     test/box-tap/errinj_set_with_enviroment_vars.test.lua
diff --git a/test/replication-luatest/CMakeLists.txt b/test/replication-luatest/CMakeLists.txt
index ffdee1b559..414adc36cb 100644
--- a/test/replication-luatest/CMakeLists.txt
+++ b/test/replication-luatest/CMakeLists.txt
@@ -41,7 +41,7 @@ foreach(test_path ${tests})
 endforeach()
 
 # See test/replication-luatest/suite.ini.
-if (CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
+if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
   set_tests_properties(
     test/replication-luatest/gh_5295_split_brain_test.lua
     test/replication-luatest/gh_6033_box_promote_demote_test.lua
```